analyze-trace.ts

```diff
@@ -328,28 +390,54 @@ async function makePrintableTree(curr: EventSpan, currentFile: string | undefine
-                return `Check file ${formatPath(currentFile!)}`;
+                treeNode.message = `Check file ${formatPath(currentFile!)}`
+                treeNode.terseMessage = `Check file ${path.basename(currentFile!)}`
+                treeNode.start = {
+                    file: currentFile!
+                }
+
+                return treeNode
+
             case "structuredTypeRelatedTo":
                 const args = event.args!;
-                return `Compare types ${args.sourceId} and ${args.targetId}`;
+                treeNode.message = `Compare types ${args.sourceId} and ${args.targetId}`;
+                treeNode.terseMessage = `Compare types ${args.sourceId} and ${args.targetId}`;
+                // TODO: Add start and end links
+                return 
+
             case "getVariancesWorker":
-                return `Compute variance of type ${event.args!.id}`;
+                treeNode.message = `Compute variance of type ${event.args!.id}`;
+                treeNode.terseMessage = `Compute variance of type ${event.args!.id}`;
+                return 
+
             default:
                 if (event.cat === "check" && event.args && event.args.pos && event.args.end) {
                     if (positionMap.has(currentFile!)) {
                         const updatedPos = positionMap.get(currentFile!)!.get(event.args.pos.toString())!;
                         const updatedEnd = positionMap.get(currentFile!)!.get(event.args.end.toString())!;
-                        return `${unmangleCamelCase(event.name)} from (line ${updatedPos[0]}, char ${updatedPos[1]}) to (line ${updatedEnd[0]}, char ${updatedEnd[1]})`;
+                        treeNode.message =  `${unmangleCamelCase(event.name)} from (line ${updatedPos[0]}, char ${updatedPos[1]}) to (line ${updatedEnd[0]}, char ${updatedEnd[1]})`;
+                        treeNode.terseMessage = unmangleCamelCase(event.name);
+                        treeNode.start = {
+                            file: currentFile!,
+                            offset: event.args.pos,
+                        }
+                        treeNode.end = {
+                            file: currentFile!,
+                            offset: event.args.end,
+                        }
+                        return treeNode;
                     }
                     else {
-                        return `${unmangleCamelCase(event.name)} from offset ${event.args.pos} to offset ${event.args.end}`;
+                        treeNode.message = `${unmangleCamelCase(event.name)} from offset ${event.args.pos} to offset ${event.args.end}`
+                        treeNode.terseMessage = unmangleCamelCase(event.name);
+                        return treeNode;
                     }
                 }
                 return undefined;
         }
     }
 
-    function updateTypeTreePositions(typeTree: any): any {
+    function updateTypeTreePositions(node: TreeNode, typeTree: any): any {
```

Comment:
Why is a `TreeNode` required?

Reply:
Passing in a root node so it can recurse, maybe I should rename it to say it will recurse